src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1603,7 +1595,7 @@ private AuthenticationResult refreshToken(final CallbackHandler callbackHandle,
                 Logger.v(TAG, "Refresh token is not returned or empty");
                 result.setRefreshToken(refreshItem.mRefreshToken);
             }
-        } catch (Exception exc) {
+        } catch (IOException exc) {
```

Comment:
@heidijinxujia we should avoid catching `IllegalArgumentException` that's why we introduced new error type for server errors to avoid any catching of this type.